file_formats/cmt_format.ml

```diff
@@ -383,6 +383,9 @@ let index_occurrences binary_annots =
       | Longident.Ldot (lid', _), Path.Pdot (path', _) ->
         reduce_and_store ~namespace lid path;
         index_components module_ lid' path'
+      | Longident.Ldot (lid', _), Path.Pextra_ty (Pdot(path', _), Pcstr_ty _) ->
+        reduce_and_store ~namespace lid path;
+        index_components module_ lid' path'
```

Comment:
You are missing the `Lapply` case. I would propose to use an `or`-pattern to make the `Path.Pextra_ty` handling more obvious:
```ocaml
      | Longident.Ldot (lid', _), (Path.Pdot (path', _) | Path.Pextra_ty (Pdot(path', _), Pcstr_ty _) )  ->
        reduce_and_store ~namespace lid path;
        index_components module_ lid' path'
      | Longident.Lapply (lid', lid''), (Path.Papply (path', path'') |  Path.Pextra_ty (Papply (path', path''), Pcstr_ty _) )  ) ->
   ```
   or you could have a `scrap_extra` function that removes `Pextra_ty` layer.

Reply:
Is the `Lapply` case actually possible ?
I tried a few things, but I failed to have a path to a constructor or a label with a `Lapply` in it. For example:
```
module F (_ : sig end) = struct
  module I = struct
    type t = A of { lbl : int } 
  end
end
module M = struct end
let _ = let open F(M) in I.A { lbl = 42 }
```
Is accepted (and indexing works), but I cannot write `let _ = F(M).I.A { lbl = 42 }` directly. Is there some convoluted case were this is actually possible ?

Reply:
The `Lapply` case is currently syntactically impossible because extended path are only allowed in type expressions.

However, this is a quite narrow line, for instance if inline record type were exposed, this would be valid
```ocaml
module F(X:sig end) = struct type t = A of { x: int } end
let f (f: F(Int).t.A) = ()
```
Thus it seems reasonable to me to be future-proof here (with a comment indicating that this case doesn't happen)?.

Reply:
That's fair, I added the case, and refactored the matching a bit in [44e0634](https://github.com/ocaml/ocaml/pull/13884/commits/44e06347e4e61a8ec61c5616aff353dc58438c45).